Makefile

```diff
@@ -20,6 +20,14 @@ IMAGE_TAG           := $(shell cat VERSION)
 CONTROL_NAMESPACE   := default
 CONTROL_KUBECONFIG  := dev/target-kubeconfig.yaml
 TARGET_KUBECONFIG   := dev/target-kubeconfig.yaml
+# Below ones are used in tests
```

Comment:
Can you also add the environment variable `MACHINE_CONTROLLER_MANAGER_DEPLOYMENT_NAME` under this section which will help while running Integration test suite locally?

Reply:
Yes sure , but I looked at how this env variable is being used to assign values, and found that its used to initialize a variable name `machineControllerManagerDeploymentName` , and lets say if env variable is not initialized then [here](https://github.com/gardener/machine-controller-manager/blob/6b9d5b4f6a87054aac63137b16ff3e36fdf946e4/pkg/test/integration/common/framework.go#L196-L198) a defualt value is assigned to `machineControllerManagerDeploymentName`. Since mcm deployment will always be named `machine-controller-manager` so I don't see much use of this env variable. WDYT?

Reply:
You're right. But there is a catch. 

`prepareMachineDeployment` method is referred in `SetupBeforeSuite` [here](https://github.com/gardener/machine-controller-manager/blob/6b9d5b4f6a87054aac63137b16ff3e36fdf946e4/pkg/test/integration/common/framework.go#L639-L649) when both Machine Controller & Machine Controller Manager images are set (in order to run within a cluster by updating the MCM deployment or by deploying an MCM on its own, based on whether the cluster is `seed` or `shoot`).

The other case, where the MCM & MC run locally [here](https://github.com/gardener/machine-controller-manager/blob/6b9d5b4f6a87054aac63137b16ff3e36fdf946e4/pkg/test/integration/common/framework.go#L651-L656) there is a `scaleMCMDeployment` called which refers `machineControllerManagerDeploymentName` variable which will throw an error if the environment variable `MACHINE_CONTROLLER_MANAGER_DEPLOYMENT_NAME ` is not set because `prepareMachineDeployment` method is never called in this scenario to set the default value.

Two things:
 - Environment variable `MACHINE_CONTROLLER_MANAGER_DEPLOYMENT_NAME ` will be needed when you are intended to run this Integration test suite locally. 
 - Or, remove the default assignment from `prepareMachineDeployment` and initialise it somewhere [here](https://github.com/gardener/machine-controller-manager/blob/6b9d5b4f6a87054aac63137b16ff3e36fdf946e4/pkg/test/integration/common/framework.go#L614) in the `SetupBeforeSuite` method.

>Since mcm deployment will always be named machine-controller-manager so I don't see much use of this env variable

Well, in Gardener ecosystem yes, it is mostly named as `machine-controller-manager`. But, MCM can operate independent of Gardener too. So, more flexibility, the better it is.

Reply:
> The other case, where the MCM & MC run locally here there is a scaleMCMDeployment called which refers machineControllerManagerDeploymentName variable which will throw an error if the environment variable MACHINE_CONTROLLER_MANAGER_DEPLOYMENT_NAME is not set because prepareMachineDeployment method is never called in this scenario to set the default value.

Oh right, then its needed , I will assign the value `machine-controller-manager` to the env variable in Makefile then.